solutions/cses-1163.mdx

```diff
@@ -61,47 +62,151 @@ int main(){
 }
 ```
 
-## Solution 2 - Reversing the steps
-
-I found another solution by reversing all the steps and calculating the maximum
-length at each step. Here is my implementation.
-
+</CPPSection>
+</LanguageSection>
+
+	
+## Solution 2 - Going Backwards
+
+**Time Complexity:** $\mathcal{O}(n)$
```

Comment:
add ## implementation header under the explanation and move time complexity to be under there.